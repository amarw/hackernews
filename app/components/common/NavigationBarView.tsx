import React from 'react';
import { View, Text, StyleProp, ViewStyle } from 'react-native';
import NavigationBarButtonView from 'app/components/common/NavigationBarButtonView';
import SeparatorLineView from 'app/components/common/SeparatorLineView';
import styles from 'app/components/common/NavigationBarView.style';

export interface NavigationBarViewProps {
  title?: string | null;
  showBackButton?: boolean;
  onBackButtonPress?: () => void;
  style?: StyleProp<ViewStyle>;
}

const NavigationBarView: React.FC<NavigationBarViewProps> = ({
  title = '',
  showBackButton = false,
  onBackButtonPress,
  ...otherProps
}: NavigationBarViewProps) => (
  <View {...otherProps}>
    <View style={styles.horizontalContainerView}>
      <View style={styles.accessoryButton}>
        {showBackButton && (
          <NavigationBarButtonView
            style={[styles.accessoryButton, { justifyContent: 'center' }]}
            textStyle={{ fontSize: 16, fontWeight: '600' }}
            onPress={onBackButtonPress}
          />
        )}
      </View>
      <Text style={styles.navigationTitle}>{title}</Text>
      <View style={styles.accessoryButton} />
    </View>
    <SeparatorLineView />
  </View>
);
export default NavigationBarView;
