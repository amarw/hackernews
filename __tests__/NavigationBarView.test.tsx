import 'react-native';
import * as React from 'react';
import NavigationBarView from '../app/components/common/NavigationBarView';

import renderer from 'react-test-renderer';

test('renders NavigationBarView correctly', () => {
    const tree = renderer.create(<NavigationBarView />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders NavigationBarView with title correctly', () => {
    const tree = renderer.create(<NavigationBarView title={'ABC'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders NavigationBarView with title & back button correctly', () => {
    const tree = renderer.create(<NavigationBarView showBackButton title={'ABC'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('simulates onBackButtonPress event correctly when NavigationBarView back button is pressed', () => {
    const mockFunction = jest.fn();
    const tree = renderer.create(
        <NavigationBarView onBackButtonPress={mockFunction} />
    );
    expect(mockFunction).not.toHaveBeenCalled();
    tree.root.props.onBackButtonPress();
    expect(mockFunction).toHaveBeenCalled();
});
