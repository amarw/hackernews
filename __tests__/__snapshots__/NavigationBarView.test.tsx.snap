// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavigationBarView correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fc6720",
          "flex": 1,
          "fontSize": 28,
          "fontWeight": "800",
          "letterSpacing": 2,
          "textAlign": "center",
          "textShadowColor": "#efeded",
          "textShadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "textShadowRadius": 1,
        }
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#cccccc",
          "height": 0.5,
          "width": undefined,
        },
      ]
    }
  />
</View>
`;

exports[`renders NavigationBarView with title & back button correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 44,
            "justifyContent": "center",
            "marginHorizontal": 8,
            "opacity": 1,
            "width": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Back
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#fc6720",
          "flex": 1,
          "fontSize": 28,
          "fontWeight": "800",
          "letterSpacing": 2,
          "textAlign": "center",
          "textShadowColor": "#efeded",
          "textShadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "textShadowRadius": 1,
        }
      }
    >
      ABC
    </Text>
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#cccccc",
          "height": 0.5,
          "width": undefined,
        },
      ]
    }
  />
</View>
`;

exports[`renders NavigationBarView with title correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fc6720",
          "flex": 1,
          "fontSize": 28,
          "fontWeight": "800",
          "letterSpacing": 2,
          "textAlign": "center",
          "textShadowColor": "#efeded",
          "textShadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "textShadowRadius": 1,
        }
      }
    >
      ABC
    </Text>
    <View
      style={
        Object {
          "height": 44,
          "marginHorizontal": 8,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        undefined,
        Object {
          "backgroundColor": "#cccccc",
          "height": 0.5,
          "width": undefined,
        },
      ]
    }
  />
</View>
`;
